cloud/scope/common.go

```diff
@@ -1,21 +1,109 @@
 package scope
 
 import (
+	"fmt"
 	"net/http"
+	"sync"
+	"time"
 
+	"github.com/bxcodec/gotcha"
+	"github.com/bxcodec/gotcha/cache"
+	"github.com/bxcodec/httpcache"
+	"github.com/bxcodec/httpcache/cache/inmem"
 	"github.com/linode/linodego"
 	"golang.org/x/oauth2"
 )
 
+const (
+	// DefaultLinodeAPITimeout is the default timeout for Linode API calls.
+	DefaultLinodeAPITimeout = 10 * time.Second
+	// DefaultLinodeAPIMaxIdleConns is the default max idle connections for Linode API calls.
+	DefaultLinodeAPIMaxIdleConns = 100
+	// DefaultLinodeAPICacheControlOverride is the default cache control override for Linode API calls.
+	DefaultLinodeAPICacheControlOverride = "max-age=60"
+	// DefaultLinodeAPICacheExpiration is the default cache expiration for Linode API calls.
+	DefaultLinodeAPICacheExpiration = 5 * time.Second
+	// DefaultLinodeAPICacheMaxSizeItem is the default cache max size item for Linode API calls.
+	DefaultLinodeAPICacheMaxSizeItem = 500
+	// DefaultLinodeAPICacheMaxMemory is the default cache max memory for Linode API calls.
+	DefaultLinodeAPICacheMaxMemory = cache.MB
+)
+
+var (
+	initClient   sync.Once
```

Comment:
This change seems to limit us to using one API client (and thus one access token) for all CAPL-managed clusters. Is there another way to have both caching for the API while supporting something like per-cluster Linode tokens?

Reply:
@cbzzz we had one access token before this change, and unnecessary client creations (access token never changes). So i think my version is better than the previous one (more configurable, and less footprint). But your point is also valid, it needs a new concept how to define per-cluster Linode token. Once we change to per-cluster, this cache solution won't be the best for us. So this is an interesting question.